tests/tests_change_settings.yml

```diff
@@ -0,0 +1,77 @@
+- hosts: all
+  tasks:
+    - name: Ensure required packages are installed
+      package:
+        name: tuned
+        state: present
+
+    - name: Ensure kernel settings profile directory exists
+      file:
+        path: /etc/tuned/kernel_settings
+        state: directory
+        mode: 0755
+
+    - name: Generate a configuration for kernel settings
+      copy:
+        src: tuned/etc/tuned/basic_settings/tuned.conf
+        dest: /etc/tuned/kernel_settings/tuned.conf
+        mode: 0644
+
+    - name: Ensure required services are enabled and started
+      service:
+        name: tuned
+        state: restarted
+        enabled: yes
+
+    - name: apply kernel_settings
+      include_role:
+        name: linux-system-roles.kernel_settings
+      vars:
+        kernel_settings_sysctl:
+          - name: fs.file-max
+            value: 400000
+          - name: kernel.threads-max
+            state: absent
+        kernel_settings_sysfs:
+          - name: /sys/kernel/debug/x86/nmi_longest_ns
+            value: 2000000
+        kernel_settings_bootloader_cmdline:
+          - name: spectre_v2
+            value: "on"
+          - name: quiet
+            state: absent
+
+    - name: reboot the machine - see if settings persist after reboot
+      reboot:
+        test_command: tuned-adm active
+
+    - name: check after reboot for ns_longest
+      shell: grep 2000000 /sys/kernel/debug/x86/nmi_longest_ns
+      
+    - name: check after reboot cmdline
+      shell: grep spectre_v2 /proc/cmdline
+
+    - name: check after reboot tuned
+      shell: tuned-adm verify -i
+
+    - name: apply kernel_settings again
+      include_role:
+        name: linux-system-roles.kernel_settings
+      vars:
+        kernel_settings_sysctl:
+          - name: fs.file-max
+            value: absent
```

Comment:
Did you mean `state: absent` i.e. you want to remove the `fs.file-max` value?  Otherwise, `absent` is not a valid value.

Reply:
Hello. It's not much clear here.
I used:
tuned/etc/tuned/basic_settings/tuned.conf
which contains:
fs.file-max=379724
so i want to reset it to default I think.

Reply:
What I should use to remove(?) value from profile?

Reply:
> What I should use to remove(?) value from profile?

Use `state: absent`.  If you have an existing tuned profile with
```
[sysctl]
fs.file-max=379724
```
and you want to remove that setting, use
```yaml
kernel_settings_sysctl:
  - name: fs.file-max
    state: absent
```
This will remove the setting from the tuned profile.  You can check the tuned profile file to see that the settings have been removed.

However, It will *not* change the system to revert the value back to its original system default value.  But this is the same as with tuned - how do you test this in tuned?  That is, when you are testing tuned, and you update a profile to remove a `sysctl` setting, and you reapply the profile, how do you verify that the `sysctl` setting was removed?

One way to check would be to rely on the fact that a system reboot is supposed to restore the value to its original setting - query the system to get the default value, save it, apply the profile to add the setting, run the role to remove the setting, reboot, query the value, verify that the value is the same as the original value
* start the test with a clean system
* `command: sysctl ...` to get the default value from the system
* apply the basic settings tuned profile - set the value to a different value
* `command: sysctl ...` or `tuned-adm verify -i` to verify that the setting is applied correctly
* run the kernel_settings role to remove the value using `state: absent` or `state: empty`
* reboot
* `command: sysctl ...` to get the value from the system - verify that this value after reboot is the same as the original starting value

Reply:
see https://github.com/linux-system-roles/kernel_settings/pull/7#discussion_r431920352

Reply:
Hello. Should be solved by last addition:
     vars:
        kernel_settings_sysctl:
          - name: fs.file-max
            value: absent
          - name: kernel.threads-max
            state: absent
        kernel_settings_bootloader_cmdline:
          - name: spectre_v2
            state: absent
          - name: quiet
            state: absent

    - name: reboot the machine - see if settings persist after reboot
      reboot:
        test_command: tuned-adm active

    - name: check after spectre_v2 reboot
      command: grep -vq spectre_v2 /proc/cmdline

    - name: check sysctl after reboot
      shell: sysctl -n fs.file-max | grep -vq 400000

Reply:
@RHEmployee there is still `value: absent` there. This sets `fs.file-max` literally to the string "absent" instead of removing it. Does not sound right.

Reply:
Hello. Thank you. So what I should use instead?

Reply:
Depends on what you want to do, but if you want to remove the setting, use `state: absent`, as mentioned already in https://github.com/linux-system-roles/kernel_settings/pull/7/files#r431920352

Reply:
Ok. Solved. Thanks.